mypy/nodes.py

```diff
@@ -2559,3 +2470,215 @@ def check_arg_names(names: List[Optional[str]], nodes: List[T], fail: Callable[[
+
+    elif isinstance(e, IndexExpr):
+        if literal(e.index) == LITERAL_YES:
+            return literal(e.base)
+
+    elif isinstance(e, NameExpr):
+        return LITERAL_TYPE
+
+    if isinstance(e, (IntExpr, FloatExpr, ComplexExpr, StrExpr, BytesExpr, UnicodeExpr)):
+        return LITERAL_YES
+
+    if literal_hash(e):
+        return LITERAL_YES
+
+    return LITERAL_NO
+
+
+def literal_hash(e: Expression) -> Optional[Key]:
+    return e.accept(hasher)
+
+
+class Hasher(ExpressionVisitor[Optional[Key]]):
+    def visit_int_expr(self, e: IntExpr) -> Key:
+        return ('Literal', e.value)
+
+    def visit_str_expr(self, e: StrExpr) -> Key:
+        return ('Literal', e.value)
+
+    def visit_bytes_expr(self, e: BytesExpr) -> Key:
+        return ('Literal', e.value)
+
+    def visit_unicode_expr(self, e: UnicodeExpr) -> Key:
+        return ('Literal', e.value)
+
+    def visit_float_expr(self, e: FloatExpr) -> Key:
+        return ('Literal', e.value)
+
+    def visit_complex_expr(self, e: ComplexExpr) -> Key:
+        return ('Literal', e.value)
+
+    def visit_star_expr(self, e: StarExpr) -> Key:
+        return ('Star', literal_hash(e.expr))
+
+    def visit_name_expr(self, e: NameExpr) -> Key:
+        return ('Var', e.name)
+
+    def visit_member_expr(self, e: MemberExpr) -> Key:
+        return ('Member', literal_hash(e.expr), e.name)
+
+    def visit_op_expr(self, e: OpExpr) -> Key:
+        return ('Binary', e.op, literal_hash(e.left), literal_hash(e.right))
+
+    def visit_comparison_expr(self, e: ComparisonExpr) -> Key:
+        rest = tuple(e.operators)  # type: Any
+        rest += tuple(literal_hash(o) for o in e.operands)
+        return ('Comparison',) + rest
+
+    def visit_unary_expr(self, e: UnaryExpr) -> Key:
+        return ('Unary', e.op, literal_hash(e.expr))
+
+    def seq_expr(self, e: Union[ListExpr, TupleExpr, SetExpr], name: str) -> Optional[Key]:
+        if all(literal(x) == LITERAL_YES for x in e.items):
+            rest = tuple(literal_hash(x) for x in e.items)  # type: Any
+            return (name,) + rest
+        return None
+
+    def visit_list_expr(self, e: ListExpr) -> Optional[Key]:
+        return self.seq_expr(e, 'List')
+
+    def visit_dict_expr(self, e: DictExpr) -> Optional[Key]:
+        if all(a and literal(a) == literal(b) == LITERAL_YES for a, b in e.items):
+            rest = tuple((literal_hash(a), literal_hash(b)) for a, b in e.items)  # type: Any
+            return ('Dict',) + rest
+        return None
+
+    def visit_tuple_expr(self, e: TupleExpr) -> Optional[Key]:
+        return self.seq_expr(e, 'Tuple')
+
+    def visit_set_expr(self, e: SetExpr) -> Optional[Key]:
+        return self.seq_expr(e, 'Set')
+
+    def visit_index_expr(self, e: IndexExpr) -> Optional[Key]:
+        if literal(e.index) == LITERAL_YES:
+            return ('Index', literal_hash(e.base), literal_hash(e.index))
+        return None
+
+    def visit_call_expr(self, e: CallExpr) -> None:
```

Comment:
This result in ` Incompatible types in assignment (expression has type Callable[[Node], None], base class "ExpressionVisitor" defined the type as Callable[[CallExpr], Tuple[Any, ...]])`
Which kinda looks like a bug to me since the returned value should be defined as `Optional[Key]`.